client/lib/analytics/sem.js

```diff
@@ -0,0 +1,121 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import cookie from 'cookie';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies.
+ */
+import { urlParseAmpCompatible } from 'lib/analytics/utils';
+
+/**
+ * Const variables.
+ */
+const debug = debugFactory( 'calypso:analytics:sem' );
+const UTM_COOKIE_MAX_AGE = 60 * 60 * 24 * 365;
+const MAX_UTM_LENGTH = 128;
+const MAX_URL_PARAM_VALUE_LENGTH = 50;
+const MAX_KEYWORD_PARAM_VALUE_LENGTH = 80;
+const MAX_GCLID_PARAM_VALUE_LENGTH = 100;
+// These are the URL params that end up in the `ad_details` cookie
+const URL_PARAM_WHITELIST = [
+	'adgroupid',
+	'campaignid',
+	'device',
+	'gclid',
+	'gclsrc',
+	'fbclid',
+	'keyword',
+	'matchtype',
+	'network',
+	'type',
+	'term',
+	'utm_campaign',
+	'utm_content',
+	'utm_medium',
+	'utm_source',
+	'utm_term',
+	'targetid', // QuanticMind
+	'locationid', // QuanticMind
+	'ref',
+	'format', // amp/non-amp
+];
+
+function isValidUtmSurceOrCampaign( value ) {
+	return null !== value.match( new RegExp( '^[a-zA-Z\\d_\\-]{1,' + MAX_UTM_LENGTH + '}$' ) );
+}
+
+function isValidOtherUrlParamValue( key, value ) {
+	if ( 'gclid' === key ) {
+		return value.length <= MAX_GCLID_PARAM_VALUE_LENGTH;
+	} else if ( 'keyword' === key ) {
+		return value.length <= MAX_KEYWORD_PARAM_VALUE_LENGTH;
```

Comment:
I'm actually leaning towards changing these limits to always mean _characters_ rather than bytes since it's a more human-friendly way to define them. Also these are quite arbitrary limits so I don't feel it's super important to be strict over their application. Defining them in bytes also penalizes some languages over others which is not totally fair. 
What do you think?

On a side note, on my Chrome both `bytes` and `chars` return the same value for me:

![image](https://user-images.githubusercontent.com/10284338/55797586-1ba36780-5acd-11e9-80c9-53d9f85d584d.png)
![image](https://user-images.githubusercontent.com/10284338/55797651-47bee880-5acd-11e9-977d-994666c79f9a.png)